minefield_game/init.cpp

```diff
@@ -35,69 +35,89 @@ namespace Init
+            std::cout << "Place mine #" << (player.mines.size() + 1) << "\n";
+            std::cout << "Enter X coordinate (1-" << maxX << "): ";
             std::cin >> x;
-            std::cout << "Ingrese coordenada Y (1-" << maxY << "): ";
+            std::cout << "Enter Y coordinate (1-" << maxY << "): ";
             std::cin >> y;
 
-            // valida que esté dentro del tablero
             if (x < 1 || x > maxX || y < 1 || y > maxY)
             {
-                std::cout << "Coordenadas fuera del tablero. Intente de nuevo.\n";
+                std::cout << "Invalid position. Try again.\n";
                 continue;
             }
 
-            // valida que no haya ya una mina ahí
-            bool exists = false;
-            for (size_t i = 0; i < player.mines.size(); i++)
+            bool validPlacement = true;
+
+            int indexX = x - 1;
+            int indexY = y - 1;
+            if (board.grid[indexY][indexX].isTaken())
             {
-                if (player.mines[i].cell.getX() == x && player.mines[i].cell.getY() == y)
+                std::cout << "That cell is taken from a previous guess. Try again.\n";
+                validPlacement = false;
+            }
+
+            for (const auto &disabled : player.disabledMineSpots)
+            {
+                if (disabled.getX() == x && disabled.getY() == y)
                 {
-                    exists = true;
+                    std::cout << "You lost a mine there before. You can't reuse that cell.\n";
+                    validPlacement = false;
                     break;
                 }
             }
-            if (exists)
+
+            for (const auto &mine : player.mines)
             {
-                std::cout << "Ya hay una mina en esa celda. Intente otra.\n";
-                continue;
+                if (mine.cell.getX() == x && mine.cell.getY() == y)
+                {
+                    std::cout << "Already placed a mine there this turn. Try again.\n";
+                    validPlacement = false;
+                    break;
+                }
             }
 
-            Mine newMine(x, y);
-            player.mines.push_back(newMine);
+            if (!validPlacement)
+                continue;
+
+            player.mines.push_back(Mine(x, y));
         }
     }
 
-    // imprime minas de un jugador
+    // prints a player's mines
     void printPlayerMines(const Player &player, int playerNumber)
     {
-        std::cout << "\nMinas del Jugador " << playerNumber << ":\n";
+        std::cout << "\nMines of the player " << playerNumber << ":\n";
         for (size_t i = 0; i < player.mines.size(); ++i)
         {
-            std::cout << "Mina #" << i + 1 << ": (" << player.mines[i].cell.getX() << "," << player.mines[i].cell.getY() << ")\n";
+            std::cout << "Mine #" << i + 1 << ": (" << player.mines[i].cell.getX() << "," << player.mines[i].cell.getY() << ")\n";

```

Comment:
#should use parenthesis on the addition, `std::cout << "Mine #" << (i + 1) << ...`